tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+  - $tarchiveID: ID of the tarchive.
+  - $dataDirBasePath: base path of the directory where all the files in table C<files>
+                      are located (i.e config value of setting 'dataDirBasePath').
+
+RETURNS: 
+
+  an array of hash references. Each hash has three keys: 'IntermedID' => ID of a file in 
+  table C<files_intermediary> , 'FileID' => ID of this file in table C<files> and 
+  'File' => absolute path of the file with the given ID.
+
+=cut
+sub getIntermediaryFilesRef {
+	my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+	
+	# This should get all files in table files_intermediary that are tied
+	# indirectly to the tarchive with ID $tarchiveId
+    my $query = 'SELECT fi.IntermedID, f.FileID, f.File FROM files_intermediary fi '
+        .       'JOIN files f ON (fi.Output_FileID=f.FileID) '
+        .       'WHERE f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)';
+
+	my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+	# Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+		$_->{'File'} = "$dataDirBasePath/$_->{'File'}" unless $_->{'File'} =~ /^\//;
+	} 
+	
+	return $filesRef;   
+}
+
+=pod
+
+=head3 getPicFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Gets the absolute paths of all the files associated to an archive 
+that are listed in table C<parameter_file> and have a parameter
+type set to C<check_pic_filename>.
+
+INPUTS:
+
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the tarchive.
+  - $dataDirBasePath: base path of the directory where all the files in table C<files>
+                      are located (i.e config value of setting 'dataDirBasePath').
+
+RETURNS: 
+
+  an array of hash references. Each hash has two keys: 'FileID' => FileID of a file 
+  in table C<parameter_file> and 'Value' => absolute path of the file with the given ID.
+
+=cut
+sub getPicFilesRef {
+	my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+	
+	# Get all files in parameter_file with parameter_type set to
+	# 'check_pic_filename' that are tied (indirectly) to the tarchive
+	# with ID $tarchiveId
+    (my $query =<<QUERY) =~ s/\s+/ /g;
+        SELECT FileID, Value FROM parameter_file pf
+	    JOIN files f USING (FileID)
+        JOIN parameter_type AS pt
+        ON (pt.ParameterTypeID=pf.ParameterTypeID)
+        WHERE pt.Name='check_pic_filename'
+	    AND (
+	            f.TarchiveSource = ? 
+	         OR f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)
+	    )
+QUERY
+
+	my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID, $tarchiveID);
+	
+	# Make sure all paths are absolute paths
+	foreach(@$filesRef) {
+		if($_->{'Value'} !~ /^\//) {
+		    $_->{'Value'} = sprintf("%s/%s/%s", $dataDirBasePath, PIC_SUBDIR, $_->{'Value'});
+	    }
+	}
+	
+	return $filesRef;
+}
+
+=pod
+
+=head3 getBackupFileName
+
+Gets the name of the tar compressed file that will contain a backup of all files
```

Comment:
Gets the name of the **backup** compressed file that will...

Reply:
Ok